api/v1alpha1/nonadminbackup_types.go

```diff
@@ -32,14 +32,19 @@ type NonAdminBackupSpec struct {
 
 	// BackupSpec defines the specification for a Velero backup.
 	BackupSpec *velerov1api.BackupSpec `json:"backupSpec,omitempty"`
-
-	// BackupStatus captures the current status of a Velero backup.
-	BackupStatus *velerov1api.BackupStatus `json:"backupStatus,omitempty"`
 }
 
 // NonAdminBackupStatus defines the observed state of NonAdminBackup
 type NonAdminBackupStatus struct {
-	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	// BackupStatus captures the current status of a Velero backup.
+
+	// +optional
+	Phase velerov1api.BackupPhase `json:"phase,omitempty"`
```

Comment:
This phase should be Non-Admin Backup CR Reconciliation phase, right ?

Reply:
or better option would be to use conditions just like we do for DPA CR status ?

Reply:
yeah this seems duplicated information. What is your goal here?

Reply:
Yes, however for this implementation I've used same as https://github.com/vmware-tanzu/velero/blob/4d548612d45d5eb0c61b2473286127dfa6e3c39d/pkg/apis/velero/v1/backup_types.go#L290-L348

It's because I think at the moment it's enough for us within `velerov1api.BackupPhase`. If we decide to have more states we can always create our own Non Admin Backup Phase.

Reply:
@mateusoliveira43 the goal was to create Phase in the Non Admin Backup separate from BackupStatus Phase, I've used the `velerov1api.BackupPhase`, but we can change it as condidions or create our own NonAdminPhase type for the Non Admin Backup.

Reply:
We do have Conditions which I've replaced, but we can stick to them:
https://github.com/migtools/oadp-non-admin/blob/master/api/v1alpha1/nonadminbackup_types.go#L42

Reply:
IMO we need the non-admin backup CR conditions as well as the whole velero backup status.

Reply:
Anyway, I will change :)

Reply:
Done.

Reply:
@mpryc Let elaborate more and give you an example why we need independent `conditions` for non-admin backup CR:
The Velero BackupStatus provides us information on the reconciliation status of Velero Backup CR but how would we know bout the reconciliation of the non-admin backup CR ? This is when the independent `conditions` on non-admin backup CR come into picture. For example if the non-admin backup controller failed to created the Velero backup and got an error, how would the user know that the request failed, in such scenario the non-admin backup controller should add a `error` condition on non-admin backup CR and specify the error on the non-admin CR status conditions so that the user gains transparency on what is actually happening with non-admin backup CR. This is in general how controllers work, `spec` is what the user wants and `status` provides information on how the CR is doing, whether its performing the intended tasks and whether the actions intended via spec are done.

Reply:
Added Conditions as well.